src/main/java/org/springframework/integration/kafka/listener/QueueingMessageListenerInvoker.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.springframework.integration.kafka.listener;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+import org.springframework.context.Lifecycle;
+import org.springframework.integration.kafka.core.KafkaMessage;
+
+/**
+ * Invokes a delegate {@link MessageListener} for all the messages passed to it, storing them
+ * in an internal queue.
+ *
+ * @author Marius Bogoevici
+ */
+public class QueueingMessageListenerInvoker implements Runnable, Lifecycle {
```

Comment:
Looks like this class should be package protected. As well as `ConsumerConnectionProvider`. They are some internal stuff for the `KafkaMessageListenerContainer`